service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/IotHubConnectionStringBuilder.java

```diff
@@ -21,23 +21,34 @@
     private static final String SHARED_ACCESS_KEY_REGEX = "^.+$";
     private static final String SHARED_ACCESS_SIGNATURE_REGEX = "^.+$";
 
-    private String hostNameSuffix;
-
     /**
-     * Static constructor to create IotHubConnectionString deserialize the given string
+     * Static constructor to create IotHubConnectionString from the given string
      *
-     * @param connectionString The serialized connection string
+     * @param connectionString The connection string
      * @return The IotHubConnectionString object
-     * @throws IOException This exception is thrown if the object creation failed
+     * @throws IOException This exception is never thrown. Users are advised to use {@link #createIotHubConnectionString(String, AuthenticationMethod)} instead
+     * since it's method signature does not declare any thrown exceptions.
+     * @deprecated Use {@link #createIotHubConnectionString(String)} since it does not declare any thrown exceptions
      */
+    @Deprecated
     public static IotHubConnectionString createConnectionString(String connectionString) throws IOException
```

Comment:
Wondering why you deprecated this and created a new one?

Reply:
The new one won't throw an IOException

Reply:
And it won't be a static "createFromConnectionString" type of constructor

Reply:
Which matches the new constructors that we are adding